eos/utils/integrate-impl.hh

```diff
@@ -114,6 +117,45 @@ namespace eos
             }
         }
     }
+
+    namespace cubature
+    {
+
+        template <size_t dim_>
+        int integrand(unsigned /* ndim */, const double *x, void *data,
+                      unsigned /* fdim */, double *fval)
+        {
+            auto& f = *static_cast<cubature::fdd<dim_> *>(data);
+            // TODO use std::experimental::to_array once it's not experimental anymore
+            std::array<double, dim_> args;
+            std::copy(x, x + dim_, args.data());
```

Comment:
Something like an ```array_view``` would be nice here. But, alas, not before C++20...

Reply:
Oh, I didn't know about it. Too bad that didn't make it into C++17